routes/ScannerExchangePage.vue

```diff
@@ -0,0 +1,422 @@
+        while(true) {
+          const options = {};
+          if(verifiablePresentation.value) {
+            options.verifiablePresentation = toRaw(
+              verifiablePresentation.value);
+            if(options.verifiablePresentation.holder) {
+              // FIXME: enable setting of other sign options such as
+              // cryptosuite / VM to use
+              options.signOptions = {
+                profileId: options.verifiablePresentation.holder
+              };
+            }
+          }
+          exchanging.value = true;
+          const {value, done} = await exchange.value.next(options);
+          exchanging.value = false;
+          // clear share-related state
+          verifiablePresentationRequest.value = undefined;
+          verifiablePresentation.value = undefined;
+          if(value?.verifiablePresentation) {
+            // set store-related state
+            const {verifiablePresentation: presentation} = value;
+            holder.value = presentation.holder;
+            const {verifiableCredential: credentials} = presentation;
+            if(!credentials) {
+              verifiableCredential.value = [];
+            } else if(!Array.isArray(credentials)) {
+              verifiableCredential.value = [credentials];
+            } else {
+              verifiableCredential.value = credentials;
+            }
+            // if there is something to store, wait for user to approve storage
+            // FIXME: enable storage of capabilities as well
+            if(verifiableCredential.value.length > 0) {
+              // user must approve store...
+              setDisplay('store');
+              await wait();
+              actionTaken = true;
+            }
+            // clear store-related state
+            verifiableCredential.value = [];
+            holder.value = '';
+          }
+          if(value?.verifiablePresentationRequest) {
+            // user must approve share...
+            setDisplay('share');
+            // set share-related state
+            verifiablePresentationRequest.value =
+              value.verifiablePresentationRequest;
+            await wait();
+            actionTaken = true;
+          }
+          // if no action was taken then nothing actionable was in in `value`
+          // (no VPR, no VP / empty VP); so set an error condition and `wait()`
+          // to let the user see something went wrong
+          if(!actionTaken) {
+            const e = new Error('Nothing was received or requested.');
+            e.name = 'OperationError';
+            await wait();
+            throw e;
+          }
+          if(done) {
+            const action = display.value === 'store' ? 'stored' : 'shared';
+            $q.notify({
+              type: 'positive',
+              message: `Successfully ${action} credential`,
+            });
+            router.push({name: 'home'});
+            // exchange is finished
+            exchange.value.close();
```

Comment:
Hmm, can this be called before pushing changing routes? We should ensure the exchange is closed first.

```suggestion
            // exchange is finished
            exchange.value.close();
            router.push({name: 'home'});
```